src/models/installation.ts

```diff
@@ -127,11 +147,26 @@ Installation.init({
 		type: DataTypes.INTEGER,
 		allowNull: true
 	}
-}, { sequelize });
+}, {
+	hooks: {
+		beforeSave: async (app: Installation, opts) => {
+			await app.encryptChangedSecretFields(opts.fields);
+			if (!app["safeSharedSecret"]) throw new Error("Fail saving app as safeSharedSecret is empty");
+		},
+		beforeBulkCreate: async (apps: Installation[], opts) => {
+			for (const app of apps) {
+				await app.encryptChangedSecretFields(opts.fields);
+				if (!app["safeSharedSecret"]) throw new Error("Fail saving app as safeSharedSecret is empty");
+			}
+		}
+	},
+	sequelize
+});
 
 export interface InstallationPayload {
 	host: string;
 	clientKey: string;
 	// secret: string;
 	sharedSecret: string;
+	safeSharedSecret: string;
```

Comment:
I'd remove this from the payload.  We want to use the same interface for both and we will be saving both from the same field. it'll make it easier to transition over.